marketplaceItems/appreciate/appResources/appData/appreciate_app.js

```diff
@@ -0,0 +1,947 @@
+            "z": 0
+        },
+        "angularDamping": 0,
+        "grab": {
+            "grabbable": false,
+            "equippableLeftRotation": {
+                "x": -0.0000152587890625,
+                "y": -0.0000152587890625,
+                "z": -0.0000152587890625,
+                "w": 1
+            },
+            "equippableRightRotation": {
+                "x": -0.0000152587890625,
+                "y": -0.0000152587890625,
+                "z": -0.0000152587890625,
+                "w": 1
+            }
+        },
+        "collisionless": true,
+        "ignoreForCollisions": true,
+        "queryAACube": {
+            "x": -0.17320507764816284,
+            "y": -0.17320507764816284,
+            "z": -0.17320507764816284,
+            "scale": 0.3464101552963257
+        },
+        "damping": 0,
+        "color": intensityEntityColorMin,
+        "clientOnly": false,
+        "avatarEntity": true,
+        "localEntity": false,
+        "faceCamera": false,
+        "isFacingAvatar": false
+    };
+    var INTENSITY_MATERIAL_ENTITY_PROPS = {
+        "name": "Intensity Entity Material",
+        "type": "Material",
+        "materialURL": "materialData"
+    };
+    var currentInitialAngularVelocity = {
+        "x": 0,
+        "y": 0,
+        "z": 0
+    };
+    function updateIntensityEntity() {
+        if (currentIntensity > 0) {
+            if (intensityEntity) {
+                intensityEntityColorMin.red = intensityEntityColorMax.red * 0.4;
+                intensityEntityColorMin.green = intensityEntityColorMax.green * 0.4;
+                intensityEntityColorMin.blue = intensityEntityColorMax.blue * 0.4;
+
+                var color = linearScaleColor(currentIntensity, intensityEntityColorMin, intensityEntityColorMax);
+
+                if (intensityMaterialEntity) {
+                    Entities.editEntity(intensityMaterialEntity, {
+                        materialData: JSON.stringify({"materialVersion": 1,
+                            "materials": [
+                                {
+                                    "roughness": 0.0,
+                                    "albedo": [color.red/255, color.green/255, color.blue/255],
+                                }
+                            ]
+                        })
+                    });
+                }
+
+                Entities.editEntity(intensityEntity, {
+                    position: halfwayBetweenHands(),
+                    dimensions: Vec3.multiply(INTENSITY_ENTITY_MAX_DIMENSIONS, currentIntensity),
+                    angularVelocity: Vec3.multiply(currentInitialAngularVelocity,
+                        currentIntensity * ANGVEL_ENTITY_MULTIPLY_FACTOR),
+                    color: color
+                });
+            } else {
+                var props = INTENSITY_ENTITY_PROPERTIES;
+                props.position = halfwayBetweenHands();
+
+                currentInitialAngularVelocity.x = Math.random() *
```

Comment:
Does Math.random() add anything special to this or just nice for initial randomness?

Reply:
the randomness IS the special part! i don't want every user's dodecahedron to be spinning at the same rate on every axis.